app/components/course-stage-dropdown.ts

```diff
@@ -0,0 +1,36 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import type CourseModel from 'codecrafters-frontend/models/course';
+import type CourseStageModel from 'codecrafters-frontend/models/course-stage';
+
+interface Signature {
+  Element: HTMLDivElement;
+
+  Args: {
+    course: CourseModel;
+    selectedCourseStage: CourseStageModel | null;
+    onAllCourseStagesDropdownLinkClick?: () => void;
+    onRequestedStageChange: (stage: CourseStageModel) => void;
```

Comment:
@ryan-gang can you explain how https://github.com/codecrafters-io/frontend/pull/1346#discussion_r1496598833 was resolved? I still see the same problem: 

<img width="842" alt="Screenshot 2024-02-21 at 14 08 25" src="https://github.com/codecrafters-io/frontend/assets/3893573/b97f5520-f07f-4fc7-9e0c-44056e6cf5ab">

Reply:
My concrete suggestion here would be to always calls this "SelectedCourseStage" in the file. So you'd rename the callback to `onSelectedCourseStageChange: (courseStage: CourseStageModel) => void`

Note that the `LanguageDropdown` component is different and needs both "selected" and "requested" because it supports states where the "requested" language isn't available and we auto-select another language.